src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_MoveFile.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.SyncNamespace
+{
+    public partial class SyncNamespaceTests : CSharpSyncNamespaceTestsBase
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task MoveFile_DeclarationNotContainedInDefaultNamespace()
+        {
+            // No "move file" action because default namespace is not container of declared namespace
+            var defaultNamespace = "A";
+            var declaredNamespace = "Foo.Bar";
+
+            var expectedFolders = new List<string[]>();
+
+            var documentPath = CreateDocumentFilePath(Array.Empty<string>(), "File1.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+namespace [||]{declaredNamespace}
+{{    
+    class Class1
+    {{
+    }}
+}}  
+        </Document>
+    </Project>
+</Workspace>";
+            await TestMoveFileToMatchNamespace(code, expectedFolders);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task MoveFile_SingleAction1()
+        {
+            // current path is <root>\
+            // expected new path is <root>\B\C\
+
+            var defaultNamespace = "A";
+            var declaredNamespace = "A.B.C";
+
+            var expectedFolders = new List<string[]>();
+            expectedFolders.Add(new[] { "B", "C" });
+
+            var documentPath = CreateDocumentFilePath(Array.Empty<string>());
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+namespace [||]{declaredNamespace}
+{{    
+    class Class1
+    {{
+    }}
+}}  
+        </Document>
+    </Project>
+</Workspace>";
+            await TestMoveFileToMatchNamespace(code, expectedFolders);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task MoveFile_SingleAction2()
+        {
+            // current path is <root>\
+            // expected new path is <root>\B\C\D\E\
+
+            var defaultNamespace = "A";
```

Comment:
not sure what this is testing. it would be interesting to test if a file already exist at the place where it want to move the file.

Reply:
This is testing when the folder already exists.

It'd be hard to test the scenario when a file with identical path already exists on disk, because test workspace and VS workspace have different behavior. Verifying it in test doesn't mean it will do the same thing in VS.